lib/acts_as_taggable_on_mongoid/models/tagging.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module ActsAsTaggableOnMongoid
+  module Models
+    # A class representing the actual tags assigned to a particular model object
+    class Tagging
```

Comment:
Copied from here:
https://github.com/mbleigh/acts-as-taggable-on/blob/master/lib/acts_as_taggable_on/taggable.rb

Lots of changes for Mongoid.

Denormalized some data into this class